python_modules/dagster/dagster/grpc/types.py

```diff
@@ -236,12 +236,24 @@ def __new__(
             ),
         )
 
-    def get_command_args(self) -> List[str]:
-        return _get_entry_point(self.pipeline_origin) + [
-            "api",
-            "execute_step",
-            serialize_dagster_namedtuple(self),
-        ]
+    def get_command_args(self, with_args=True) -> List[str]:
+        """
+        Get the command args to run this step. If with_args is False, then get_command_env should
+        be used to pass the args over env vars.
+        """
+        return (
+            _get_entry_point(self.pipeline_origin)
+            + ["api", "execute_step"]
+            + [serialize_dagster_namedtuple(self)]
+            if with_args
+            else []
```

Comment:
i find this hard to read what exactly the if is gating, maybe extra parens or be more explicit